### PR TITLE
Move get planning scene service functionality into PSM for reusability

### PR DESCRIPTION
**Pull request by davetcoleman**
This will eliminate that pesky warning message we've had for years:

> ros.moveit_ros_planning.planning_scene_monitor: Failed to call service get_planning_scene, have you launched move_group? at /moveit/moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp:489

All you need to do is start the service for your master PSM.

### CODE REVIEW

File: moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h [hunk 1]
```diff
@@ -337,6 +338,13 @@ class PlanningSceneMonitor : private boost::noncopyable
    */
   bool requestPlanningSceneState(const std::string& service_name = DEFAULT_PLANNING_SCENE_SERVICE);
 
+  /** @brief Create an optional service for getting the complete planning scene
+   *         This is useful for satisfying the Rviz PlanningScene display's need for a service
+   *         without having to use MoveGroup
+   *  @param service_name The topic to provide the service
+   */
+  void providePlanningSceneService(const std::string& service_name = DEFAULT_PLANNING_SCENE_SERVICE);
```
Comment (henningkayser) [inline moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h]: Wouldn't the default mean that we start a service that the service client in the same PSM instance would subscribe to?

Comment (davetcoleman) [reply moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h]: I've just added a new commit with lots of explanations and warnings about this scenario.

Overall, yes, PSM is now capable of being a client or a server, but you shouldn't be both.

**Comment by rhaschke** [comment]
@henningkayser: Why did you merge-committed this? The second commit is just a cleanup.

**Comment by rhaschke** [comment]
I also had some more comments after Dave's 2nd commit...

**Comment by rhaschke** [comment]
I force-pushed master to revert this PR for now. I will open a new PR.

**Comment by davetcoleman** [comment]
did you have to revert it? Seems like a fixup PR would have sufficed...

**Comment by rhaschke** [comment]
Henning didn't obey to our rule to squash-commit fixups and thus I took the chance to cleanup all commits.

**Pull request merged**